src/python/pants/option/global_options.py

```diff
@@ -251,11 +251,59 @@ def from_options(
         )
 
 
+@dataclass(frozen=True)
+class LocalStoreOptions:
+    """A collection of all options related to the local store.
+
+    TODO: These options should move to a Subsystem once we add support for "bootstrap" Subsystems (ie,
+    allowing Subsystems to be consumed before the Scheduler has been created).
+    """
+
+    store_dir: str
+    processes_max_size_bytes: int
+    files_max_size_bytes: int
+    directories_max_size_bytes: int
+
+    def target_total_size_bytes(self) -> int:
+        """Returns the target total size of all of the stores.
+
+        The `max_size` values are caps on the total size of each store: the "target" size
+        is the size that garbage collection will attempt to shrink the stores to each time
+        it runs.
+
+        NB: This value is not currently configurable, but that could be desirable in the future.
+        """
+        max_total_size_bytes = (
+            self.processes_max_size_bytes
+            + self.files_max_size_bytes
+            + self.directories_max_size_bytes
+        )
+        return max_total_size_bytes // 10
+
+    @classmethod
+    def from_options(cls, options: OptionValueContainer) -> LocalStoreOptions:
+        return cls(
+            store_dir=Path(options.local_store_dir).resolve().as_posix(),
```

Comment:
I've not understood our use of `as_posix` in contexts where we don't actually care if the separator is `/`. Doesn't it make more sense to just `str(...)` the Path if just to not make an unlikely Windows port easier?